Client/Application/AppState.swift

```diff
@@ -0,0 +1,29 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+protocol AppStateDelegate: class {
+    func appDidUpdateState(appState: AppState)
+}
+
+enum AppState {
+    case Tab(tabState: TabState)
```

Comment:
I really don't like this being a enum which is peppered throughout the `MenuView` code.